pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -34,17 +40,29 @@
 import org.apache.pulsar.io.core.Source;
 import org.jboss.util.Classes;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
-
 @Slf4j
 public class PulsarSource<T> implements Source<T> {
 
     private PulsarClient pulsarClient;
     private PulsarSourceConfig pulsarSourceConfig;
-    private Map<String, SerDe> topicToSerDeMap = new HashMap<>();
+    private Map<String, SerDe<T>> topicToSerDeMap = new HashMap<>();
+
+    private final Schema<T> emptySchema = new Schema<T>() {
```

Comment:
Why is this needed?

Reply:
Because of the aforementioned issue raised by @merlimat; a Consumer must be created with a schema so I needed something to capture the type before I added the individual topic/schema pairs.

Reply:
is there a way for us the avoid this?  This can be confusing for the reader.